pkgs/development/python-modules/hachoir/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, urwid
+, wxPython_4_0
```

Comment:
Does Nix support optional/recommended dependencies? wxPython is only needed for hachoir-wx. Most hachoir programs are text-only, to be used in the command line.

Reply:
I'm aware of that. Maybe hachoir-wx should be made into a separate nix package then. I'd like to avoid installing hachoir-wx and then have it raise an exception if wxPython is not installed. Same for havoir-urwid. I'll look into how optional dependencies are handled by nix.

Reply:
Is there a way to install hachoir without hachoir-wx and hachoir-urwid? There seems to be a way to [enable optional dependencies](https://github.com/NixOS/nixpkgs/blob/ed54a5b51dc9542df94f70b25a13d86d1f494e64/pkgs/applications/networking/newsreaders/pan/default.nix) using build flags. Separate derivations for hachoir-wx and hachoir-urwid would be another option.